src/UserPullPieChart.tsx

```diff
@@ -0,0 +1,89 @@
+import { Paper } from '@mui/material';
+import {
+  Cell,
+  Legend,
+  Pie,
+  PieChart,
+  ResponsiveContainer,
+  Tooltip,
+} from 'recharts';
+import { PullRequestKeyMetrics } from './types';
+
+const COLORS = [
+  '#e60049',
+  '#0bb4ff',
+  '#50e991',
+  '#e6d800',
+  '#9b19f5',
+  '#ffa300',
+  '#dc0ab4',
+  '#b3d4ff',
+  '#00bfa0',
+];
+type Props = {
+  authoredPullRequests: PullRequestKeyMetrics[];
+};
+
+function getPullPieData(pullRequests: PullRequestKeyMetrics[]) {
+  const pullAuthorCounter: any = {};
+  let colorCounter = 0;
+  pullRequests.forEach((pull) => {
+    if (pullAuthorCounter[pull.author]) {
+      pullAuthorCounter[pull.author].value =
+        pullAuthorCounter[pull.author].value + 1;
+      pullAuthorCounter[pull.author].totalCodeChanges =
+        pullAuthorCounter[pull.author].totalCodeChanges + pull.totalCodeChanges;
+    } else {
+      pullAuthorCounter[pull.author] = {
+        value: 1,
+        totalCodeChanges: pull.totalCodeChanges,
+        color: COLORS[colorCounter],
+      };
+      colorCounter++;
+    }
+  });
+
+  const authorPieData = Object.keys(pullAuthorCounter).map((key) => {
+    return {
+      name: key,
```

Comment:
If we assign `name: key` on L37 we could we get away with `Object.entries(pullAuthorCounter)` here?